### PR TITLE
Added blank Wireguard service

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/29928

**No functional changes**, added blank BraveVpnWireguardService with all infrastructure files. It is not called from anywhere, just added as dependency to be built. In order to simplify review process of Wireguard integration all further logic will be added as later separte PRs according to the plan https://docs.google.com/document/d/18HMN-G4NAxrdaihC-ybl9-blhCqT7Vca-E1TdK2zJls/edit#bookmark=id.4julw3f5kk3u

- Added a new service next to brave vpn helper service.
- Added `win_build_output/midl/brave` files because the service will be managed via COM interfaces.
- Updated build infrastructure and build configs to work with new files correctly

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- No tests required

### CODE REVIEW

File: .gitattributes
Comment (mherrmann) [inline .gitattributes:53]: This can be replaced by `win_build_output/midl/brave/**/*.c`:

```
mkdir play-with-gitattributes
cd play-with-gitattributes
git init
mkdir a
mkdir a\b
mkdir a\b\c
mkdir a\b\c\d
echo "hi" > a\b\c\d\e.txt
echo f > f.txt
git add -A
git commit -m "Add some files"
echo a/**/*.txt foo=bar > .gitattributes
git check-attr foo a\b\c\d\e.txt f.txt
```

gives:

```
"a\\b\\c\\d\\e.txt": foo: bar
f.txt: foo: unspecified
```

File: build/commands/lib/util.js
Comment (mherrmann) [inline build/commands/lib/util.js:489]: I would replace this duplicate code by a loop over `["google_update", "brave"]`.

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn
Comment (mherrmann) [inline components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn:19]: I don't doubt that this is correct, but I wonder why. Didn't we stop supporting Windows 7 at the beginning of the year?

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn
Comment (mherrmann) [inline components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn:55]: This line seems to be duplicate.

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/brave_wireguard_manager_idl.idl
Comment (mherrmann) [inline components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/brave_wireguard_manager_idl.idl:20]: Should this be indented more?

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/service_utils.cc
Comment (mherrmann) [inline components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/service_utils.cc:21]: This function is defined already in several files. Can we extract it to remove the duplicate code?

**Review by mherrmann — CHANGES_REQUESTED** [review]

**Comment by spylogsster** [comment]
Moving to draft for some time

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn
Comment (spylogsster) [reply components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn]: Im not sure, just did same as elevation service

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn
Comment (spylogsster) [reply components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn]: fixed

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/brave_wireguard_manager_idl.idl
Comment (spylogsster) [reply components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/brave_wireguard_manager_idl.idl]: done

File: build/commands/lib/util.js
Comment (spylogsster) [reply build/commands/lib/util.js]: done

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/service_utils.cc
Comment (spylogsster) [reply components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/service_utils.cc]: done

File: components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn
Comment (mherrmann) [reply components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn]: https://support.brave.com/hc/en-us/articles/11197967945613-Sunsetting-support-for-Windows-7-and-8-8-1

I do see the same code in `elevation_service` upstream. But I suspect it's outdated and will be removed soon.

I would try to remove the `config("winver")` block and see if that removal causes problems.

Comment (spylogsster) [reply components/brave_vpn/browser/connection/win/brave_vpn_wireguard_service/BUILD.gn]: done

**Review by mherrmann — APPROVED** [review]

**Review by bridiver — APPROVED** [review]
approved for code owner only (util.js). Please get approval from @goodov or @iefremov before merge

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_main.h
Comment (goodov) [inline components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_main.h:9]: remove/replace?

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_utils.h
Comment (goodov) [inline components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_utils.h:9]: ?

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/brave_wireguard_manager.h
Comment (goodov) [inline components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/brave_wireguard_manager.h:12]: remove?

File: components/brave_vpn/browser/connection/common/win/utils.h
Comment (spylogsster) [reply components/brave_vpn/browser/connection/common/win/utils.h]: a6ed731f4d5d92f0667df8d1ca897155a0892c52

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_main.h
Comment (spylogsster) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_main.h]: done

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/brave_wireguard_manager.h
Comment (spylogsster) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/brave_wireguard_manager.h]: done

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_main.h
Comment (spylogsster) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_main.h]: done

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.h
Comment (goodov) [inline components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.h:14]: I assume we want our executable to have `brave_vpn` in the name? If that's the case, then let's have all related variables also include "Vpn" in the name:
`kBraveWireguardServiceExecutable` -> `kBraveVpnWireguardServiceExecutable`

It that's not the case, then we can name the executable `brave_wireguard_service`, and use `brave_wireguard` prefix everywhere. Right now we have `LIBID_BraveVpnWireguardServiceLib`, but `IID_IBraveWireguardManager` which is not very consistent.

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.h
Comment (goodov) [inline components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.h:23]: `BraveWireguardService` -> `BraveVpnWireguardService` in all these functions (if we're keeping executable name `brave_vpn_wireguard_service.exe`).

File: components/brave_vpn/browser/connection/common/win/BUILD.gn
Comment (spylogsster) [reply components/brave_vpn/browser/connection/common/win/BUILD.gn:9]: added

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/brave_vpn_wireguard_service.ver
Comment (spylogsster) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/brave_vpn_wireguard_service.ver]: renamed

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/brave_wireguard_service_crash_reporter_client.cc
Comment (spylogsster) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/brave_wireguard_service_crash_reporter_client.cc]: renamed

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.cc
Comment (spylogsster) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.cc]: done

File: chromium_src/components/crash/core/app/crashpad.cc
Comment (spylogsster) [reply chromium_src/components/crash/core/app/crashpad.cc]: renamed

File: components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.h
Comment (spylogsster) [reply components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/service_constants.h]: renamed

File: chromium_src/components/crash/core/app/crashpad.cc
Comment (goodov) [inline chromium_src/components/crash/core/app/crashpad.cc:18]: sorry to changing this again, but I think this should be `brave-vpn-wireguard-service` to be in sync with the actual executable name.

File: components/brave_vpn/browser/connection/common/win/BUILD.gn
Comment (goodov) [reply components/brave_vpn/browser/connection/common/win/BUILD.gn:9]: huh. I was expecting visibility here to only have `"//brave/components/brave_vpn/*"`.
`ScHandleTraits` looks like a `//base` thing which is copy-pasted few times in chromium :/

**Review by goodov — APPROVED** [review]
lgtm, but pls change `kBraveWireguardProcessType` to `brave-vpn-wireguard-service` to be in sync with exe name.

File: chromium_src/components/crash/core/app/crashpad.cc
Comment (spylogsster) [reply chromium_src/components/crash/core/app/crashpad.cc]: done

**Pull request merged**
